paddle/fluid/inference/api/analysis_predictor.cc

```diff
@@ -185,10 +185,49 @@ bool AnalysisPredictor::PrepareExecutor() {
   return true;
 }
 
+void AnalysisPredictor::MkldnnPreSet(const std::vector<PaddleTensor> &inputs) {
+#ifdef PADDLE_WITH_MKLDNN
+  VLOG(2) << "AnalysisPredictor::Run get_cur_mkldnn_session_id="
+          << platform::get_cur_mkldnn_session_id();
+  // In cache clearing mode.
+  if (config_.mkldnn_cache_capacity_ > 0) {
```

Comment:
There is confusion which one decides cache clearing mode, session_id or cache_capacity?
I think session_id is better, cache_capacity is just one property of session_id.

Reply:
`config_.mkldnn_cache_capacity_` is specified by the users, but `session_id` is used internally.
> which one decides cache clearing mode

Thus, for users, `config_.mkldnn_cache_capacity_` decides the cache clearing mode. 

In next PRs, I will move `paddle::platform::set_cur_mkldnn_session_id()`, `platform::set_cur_input_shape_cache_capacity(0)`,`platform::set_cur_input_shape_str("")` into `MKLDNNDeviceContext` class function. Or create a `MKLDNNDeviceContext::Reset()` function. i.e
```
void MKLDNNDeviceContext::Reset() {
  MKLDNNDeviceContext::set_cur_mkldnn_session_id(
        MKLDNNDeviceContext::kMKLDNNSessionID_Default);
    MKLDNNDeviceContext::set_cur_input_shape_cache_capacity(0);
    MKLDNNDeviceContext::set_cur_input_shape_str("");
}
```
And in `analysis_predictor.cc`, we will call
```
if (config_.mkldnn_cache_capacity_ > 0) {
  auto &pool = platform::DeviceContextPool::Instance();
  auto *dev_ctx = dynamic_cast<platform::MKLDNNDeviceContext *>(
      pool.Get(platform::CPUPlace()));
  dev_ctx->Reset();
}
```

Reply:
so you mean from user perspective, there are only 2 modes, one is normal mode with default cache_capacity setting, the other is cache clearing mode with explicit cache_capacity setting, session_id, shape_str are just internal variables used for implementation. right?

Reply:
Yes, you are right.

Reply:
LGTM